v2/carbs/carbs.go

```diff
@@ -0,0 +1,300 @@
+package carbs
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	bs "github.com/ipfs/go-ipfs-blockstore"
```

Comment:
I'd use a slightly better name here, like blockstore; a global "bs" name is too short to be intuitive